File: Count and Say.java
public class Solution {
    public String countAndSay(int n) {
        String temp = "1";
        for (int loop = 1; loop < n; loop++) {
            StringBuilder result = new StringBuilder();
            Character digit = temp.charAt(0);
            int count = 0;
            for (int i = 0; i < temp.length(); i++) {
                if (digit == temp.charAt(i)) {
                    count++;
                }
                else {
                    result.append(count);
                    result.append(digit);
                    digit = temp.charAt(i);
                    count = 1;
                }
            }
            result.append(count);
            result.append(digit);
            temp = result.toString();
        }
        return temp;
    }
}

File: Count and Say.cpp

class Solution {
public:
    string countAndSay(int n) {
        string result = "1";
        for (int c = 1; c < n; c++) {
			string temp = result;
			result = "";
            char digit = temp[0];
            int count = 0;
            for (int i = 0; i < temp.length(); i++) {
                if (digit == temp[i]) {
                    count++;
                }
                else {
                    result += to_string(count);
                    result += digit;
                    digit = temp[i];
                    count = 1;
                }
            }
            result += to_string(count);
            result += digit;
        }
        return result;
    }
};